src/Tribe/Utils/Post_Root_Pool.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+
+class Tribe__Utils__Post_Root_Pool {
+
+	/**
+	 * @var string
+	 */
+	protected $pool_transient_name = 'tribe_ticket_prefix_pool';
+
+	/**
+	 * @var array|bool
+	 */
+	protected static $prefix_pool = false;
+
+	/**
+	 * @var string
+	 */
+	protected $root_separator = '-';
+
+	/**
+	 * @var array
+	 */
+	protected $postfix = 1;
+
+	/**
+	 * Generates a unique root for a post using its post_name.
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return string
+	 */
+	public function generate_unique_root( WP_Post $post ) {
```

Comment:
Ah OK, I _think_ I got my answer over here:

[Tribe__Tickets_Plus__Meta__Unique_ID::get_event_root()](https://github.com/moderntribe/event-tickets-plus/pull/147/files#diff-9c31cd9ece5407537804d45d92c59d73R82)

So if the responsibility for preventing multiple unique roots per post is on the calling function maybe add a note in the docblock or even mark the method `@internal`.